src/main/java/net/dv8tion/jda/internal/managers/GuildScheduledEventManagerImpl.java

```diff
@@ -0,0 +1,231 @@
+        if (isPermissionChecksEnabled())
+            checkPermissions();
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEvent getGuildScheduledEvent()
+    {
+        GuildScheduledEvent realEvent = event.getGuild().getScheduledEventById(event.getIdLong());
+        if (realEvent != null)
+            event = realEvent;
+        return event;
+    }
+
+    @Nonnull
+    @Override
+    @CheckReturnValue
+    public GuildScheduledEventManagerImpl setName(@Nonnull String name)
+    {
+        Checks.notBlank(name, "Name");
+        Checks.notLonger(name, GuildScheduledEvent.MAX_NAME_LENGTH, "Name");
+        this.name = name;
+        set |= NAME;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventManager setDescription(@Nullable String description)
+    {
+        Checks.notLonger(description, GuildScheduledEvent.MAX_DESCRIPTION_LENGTH, "Description");
+        this.description = description;
+        set |= DESCRIPTION;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventManager setImage(@Nullable Icon icon)
+    {
+        this.image = icon;
+        set |= IMAGE;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventManager setLocation(@Nonnull GuildChannel channel)
+    {
+        Checks.notNull(channel, "Channel");
+        if (!channel.getGuild().equals(event.getGuild()))
+        {
+            throw new IllegalArgumentException("Invalid parameter: Channel has to be from the same guild as the scheduled event!");
+        }
+        else if (channel instanceof StageChannel)
+        {
+            this.channelId = channel.getIdLong();
+            this.entityType = GuildScheduledEvent.Type.STAGE_INSTANCE;
+        }
+        else if (channel instanceof VoiceChannel)
+        {
+            this.channelId = channel.getIdLong();
+            this.entityType = GuildScheduledEvent.Type.VOICE;
+        }
+        else
+        {
+            throw new IllegalArgumentException("Invalid parameter: Can only set location to Voice and Stage Channels!");
+        }
+
+        set |= LOCATION;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventManager setLocation(@Nonnull String location)
+    {
+        Checks.notBlank(location, "Location");
+        Checks.notLonger(location, GuildScheduledEvent.MAX_LOCATION_LENGTH, "Location");
+        this.location = location;
+        this.entityType = GuildScheduledEvent.Type.EXTERNAL;
+        set |= LOCATION;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventManager setStartTime(@Nonnull TemporalAccessor startTime)
+    {
+        Checks.notNull(startTime, "Start Time");
+        Checks.check(Helpers.toOffsetDateTime(startTime).isBefore(OffsetDateTime.now().plusYears(5)), "Scheduled start and end times must be within five years.");
+        this.startTime = Helpers.toOffsetDateTime(startTime);
```

Comment:
Move startTime into a variable to avoid converting it twice, same with the one in end time